From 7f40865b4528d0275e6aaa64b13191834ed08699 Mon Sep 17 00:00:00 2001
From: Vincent Bernat <vincent@bernat.im>
Date: Sat, 28 Apr 2018 17:38:48 +0200
Subject: [PATCH] qa: FakeNeighbor should have an extended_message attribute

Otherwise, we fail here:

    + env INTERPRETER=python2.7 ETC=/build/exabgp-4.0.6/etc/exabgp exabgp_log_enable=false python2.7 -m nose ./qa/tests/cache_test.py ./qa/tests/control_test.py ./qa/tests/decode_test.py ./qa/tests/flow_test.py ./qa/tests/l2vpn_test.py ./qa/tests/open_test.py ./qa/tests/parsing_test.py
    ......EE.........
    ======================================================================
    ERROR: test_decoding_udpate_asn (decode_test.TestUpdateDecoding)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/build/exabgp-4.0.6/qa/tests/decode_test.py", line 356, in setUp
        capa = Capabilities().new(neighbor,False)
      File "/build/exabgp-4.0.6/lib/exabgp/bgp/message/open/capability/capabilities.py", line 145, in new
        self._extended_message(neighbor)
      File "/build/exabgp-4.0.6/lib/exabgp/bgp/message/open/capability/capabilities.py", line 119, in _extended_message
        if not neighbor.extended_message:
    AttributeError: 'FakeNeighbor' object has no attribute 'extended_message'

    ======================================================================
    ERROR: test_decoding_udpate_asn4 (decode_test.TestUpdateDecoding)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/build/exabgp-4.0.6/qa/tests/decode_test.py", line 356, in setUp
        capa = Capabilities().new(neighbor,False)
      File "/build/exabgp-4.0.6/lib/exabgp/bgp/message/open/capability/capabilities.py", line 145, in new
        self._extended_message(neighbor)
      File "/build/exabgp-4.0.6/lib/exabgp/bgp/message/open/capability/capabilities.py", line 119, in _extended_message
        if not neighbor.extended_message:
    AttributeError: 'FakeNeighbor' object has no attribute 'extended_message'

    ----------------------------------------------------------------------
    Ran 17 tests in 1.986s
---
 qa/tests/decode_test.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/qa/tests/decode_test.py b/qa/tests/decode_test.py
index 8faf1f531732..7a4eb842d41f 100755
--- a/qa/tests/decode_test.py
+++ b/qa/tests/decode_test.py
@@ -317,6 +317,7 @@ class FakeNeighbor (object):
 	hold_time = HoldTime(180)
 	asn4 = False
 	add_path = 0
+	extended_message = False
 
 	# capability
 	route_refresh = False
-- 
2.17.0

